toolchain/check/testdata/builtin_conversions/no_prelude/fail_convert_class_type_to_generic_facet_value.carbon

```diff
@@ -0,0 +1,265 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/fail_convert_class_type_to_generic_facet_value.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/fail_convert_class_type_to_generic_facet_value.carbon
+
+interface Generic(Scalar:! type) {
+  fn F();
+}
+
+class GenericParam {}
+class WrongGenericParam {}
+
+class ImplsGeneric {}
+impl ImplsGeneric as Generic(GenericParam) {
+  fn F() {}
+}
+
+fn CallGenericMethod(T:! type, U:! Generic(T)) {}
+
+fn G() {
+  // CHECK:STDERR: fail_convert_class_type_to_generic_facet_value.carbon:[[@LINE+7]]:3: error: `Core.ImplicitAs` implicitly referenced here, but package `Core` not found [CoreNotFound]
+  // CHECK:STDERR:   CallGenericMethod(WrongGenericParam, ImplsGeneric);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Is this the error you were expecting here? Maybe this shouldn't be a `no_prelude` test.

Reply:
Yes and no. With the prelude you get "unable to convert, does not implement ImplicitAs...". But that requires adding the whole prelude to the test. I saw that we have other no_prelude tests that have this error in them as well in similar situations (example: https://github.com/carbon-language/carbon-lang/blob/a735a4e463ee638e299cc3061198bd9d487bebc8/toolchain/check/testdata/var/no_prelude/fail_init_type_mismatch.carbon#L12).

It does require this extra secret knowledge that this error means it tried to do a conversion we were not expecting it to attempt at all. But it makes debugging so much better. Do you have an opinion on which way to go, given all of that?

Reply:
I'm a little uncomfortable that the exact failure path we're testing isn't one that we expect developers to actually encounter. We also might end up moving some of the builtin conversions into the prelude over time, which might mean this ends up testing less than we intend. But that's a bit hypothetical and I don't feel strongly about it.

Hm, an alternative would be to add a "mini prelude" that contains only `ImplicitAs` to this test, eg:
```
// --- core.carbon

package Core api;

interface ImplicitAs(T:! type) {
  fn Convert[self: Self]() -> T;
}

// --- your_test.carbon

import Core;
```
... should work. I'm not sure if it's worth doing that in this case.

Reply:
Right, yeah, I was doing that when I was debugging as well! But removed it since I saw this error elsewhere and I was hesitant about creating work for future people if this interface might change. Sounds like this isn't worrying you so much, so I will take that approach I guess.